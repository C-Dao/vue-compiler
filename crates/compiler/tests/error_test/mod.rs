use super::common::{serialize_yaml, get_errors};
use insta::assert_snapshot;

macro_rules! assert_yaml {
    ($val: expr, $expr: expr) => {
        let name = insta::_macro_support::AutoName;
        let val = serialize_yaml($val);
        assert_snapshot!(name, val, $expr);
    };
}

#[test]
fn test_scan() {
    let cases = [
        // r#"a {{ b "#,
        // r#"<![CDATA["#,
        // r#"{{}}"#,
        // r#"{{test}}"#,
        // r#"<a test="value">...</a>"#,
        // r#"<a v-bind:['foo' + bar]="value">...</a>"#,
        // r#"<tag =value />"#,
        // r#"<a =123 />"#,
        // r#"<a ==123 />"#,
        // r#"<a b="" />"#,
        // r#"<a == />"#,
        // r#"<a wrong-attr>=123 />"#,
        // r#"<a></a < / attr attr=">" >"#,
        // r#"<a attr="1123"#,              // unclosed quote
        // r#"<a attr=""#,                  // unclosed without val
        // r#"<!-->"#,                      // abrupt closing
        // r#"<!--->"#,                     // abrupt closing
        // r#"<!---->"#,                    // ok
        // r#"<!-- nested <!--> text -->"#, // ok
        // r#"<p v-err=232/>"#,
    ];
    for case in cases {
        let val: Vec<_> = get_errors(case);
        assert_yaml!(val, case);
    }
}

#[test]
fn test_abrupt_closing_of_comment() {
    let cases = [
        r#"<template><!--></template>"#,
        r#"<template><!---></template>"#,
        r#"<template><!----></template>"#,
    ];
    for case in cases {
        let val: Vec<_> = get_errors(case);
        assert_yaml!(val, case);
    }
}
