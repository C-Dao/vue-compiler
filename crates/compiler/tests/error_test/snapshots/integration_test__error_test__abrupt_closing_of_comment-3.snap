---
source: crates/compiler/tests/error_test/mod.rs
assertion_line: 8
expression: "<template><!----></template>"

---
[]

